### DISCUSSION TITLE
Updating Compiler to Latest Version

### DISCUSSION THREAD

**Discussion by AMOfficialGithub** [dotnet/roslyn#49117] [Q&A]
Hi everyone, I've been doing C# for a month now but I have a question with the current compiler I have. Even though I have the latest .NET package for this year, the compiler i'm using is for version 5 which isn't the latest.

How do I update the compiler so that it will work with the latest C# language? I got here from a notification from the CLI but i'm not understanding the steps I need to take to have the latest working compiler.

**Comment by PathogenDavid** [comment]
It sounds like you're using the compiler that comes with Windows (IE: `C:\Windows\Microsoft.NET\Framework64\v4.0.30319\csc.exe`), that compiler is the old compiler from the .NET Framework days before Roslyn exist. (Frequently referred to as the "native compiler".)

Most developers use the C# compiler through MSBuild using an IDE like Visual Studio or VSCode.

If you don't want to use an IDE, you're probably better off using the `dotnet` command line. [Getting started with .NET](https://docs.microsoft.com/en-us/dotnet/core/get-started) shows some basic usage, but as [the official getting started tutorials](https://docs.microsoft.com/en-us/dotnet/standard/get-started) push you towards an IDE really quick.

----------------------

If you really just want to invoke the compiler directly on the command line, you have a few options for the modern compiler.

If you have Visual Studio installed, it comes with a more version of Roslyn. It's most easily accessed using the [Developer Command Prompt](https://docs.microsoft.com/en-us/dotnet/framework/tools/developer-command-prompt-for-vs). But you can find it at `C:\Program Files (x86)\Microsoft Visual Studio\<Version>\<Edition>\MSBuild\Current\Bin\Roslyn\csc.exe` (substitute your version and edition as appropriate.) Note that this version of Roslyn will be dependent on you updating Visual Studio, not the .NET SDK.

If you want to run the version that comes with the .NET SDK, that's slightly less straightforward because it isn't distributed as an executable. The compiler is located at `C:\Program Files\dotnet\sdk\<SDK version>\Roslyn\bincore\csc.dll` You can invoke it with the dotnet CLI. For example, for the .NET 3.1.403 SDK you can run `dotnet "C:\Program Files\dotnet\sdk\3.1.403\Roslyn\bincore\csc.dll"`:

![image](https://user-images.githubusercontent.com/278957/97913379-d03a5500-1d13-11eb-9490-d8079394ab8b.png)
